lark/load_grammar.py

```diff
@@ -1009,6 +1015,15 @@ def load_grammar(self, grammar_text, grammar_name='<?>', import_paths=[]):
             term_defs += new_td
             rule_defs += new_rd
 
+        for r in overriding_rules:
+            name = r[0]
+            overridden, rule_defs = classify_bool(rule_defs, lambda r: r[0] == name)    # FIXME inefficient
```

Comment:
I think a small comment explaining what is happening here would help. It took me a moment to realize that this throws away the `overriden` rules.

Also, can't we do the check at a later point when we are iterating over `rule_defs` anyway? But that would probably be less clear so I am not sure that it is a good idea.

Reply:
Maybe there's a better way to do this, but I think it would require changing a lot of how that function operates. And it's probably best to separate new features and big refactors, if possible.

Reply:
Anyway, added comments.

Reply:
> And it's probably best to separate new features and big refactors, if possible.

Agreed. I wasn't even sure how to do this correctly. I was just saying that it took me a moment that this was the place where the old definition is removed.